src/main/java/uk/co/ramp/event/ContactEventProcessor.java

```diff
@@ -59,19 +64,31 @@ private boolean isContactIsolated(
 
   Optional<InfectionEvent> evaluateContact(ContactEvent contacts, double proportionInfectious) {
     int time = contacts.time();
-    Case potentialSpreader = population.get(contacts.to());
-    Case victim = population.get(contacts.from());
+
+    Case potentialSpreader =
+        getMostSevere(population.get(contacts.to()), population.get(contacts.from()));
+    Case victim =
+        potentialSpreader == population.get(contacts.to())
+            ? population.get(contacts.from())
+            : population.get(contacts.to());
 
     boolean shouldIsolateContact =
         isContactIsolated(contacts, potentialSpreader, victim, proportionInfectious, time);
 
-    if (shouldIsolateContact) {
+    boolean compliesWithIsolation = true;
+    // check for whether infector would isolate
+    if (distributionSampler.uniformBetweenZeroAndOne() < potentialSpreader.isolationCompliance()) {
+      LOGGER.info("person with id %d is not complying with isolation policy for this contact.");
+      compliesWithIsolation = false;
+    }
+
+    if (shouldIsolateContact && compliesWithIsolation) {
```

Comment:
Isolation compliance has already been implemented. If you have a look at `SingleCaseIsolationPolicy.java` around line 185, you will see that the `isContactIsolated()` call in from this file on line 75 already captures this.

We were previously using the 'compliance', as opposed to 'isolationCompliance', but you have implemented the name change correctly for that in this PR. I think the check here should be removed though

Reply:
ah fair enough I'll remove that then.